adapters/jwplayer/jwplayer_test.go

```diff
@@ -0,0 +1,183 @@
+
+	assert.Len(t, resultJSON.Imp, 1, "Imp count should be equal or less than Imps from input. In this test, should be 1.")
+	assert.Empty(t, resultJSON.Imp[0].Ext, "Ext should be deleted")
+	assert.Equal(t, "test_placement_id" , resultJSON.Imp[0].TagID, "placement id should be set to TagID")
+	assert.Equal(t, int(resultJSON.Imp[0].Video.H), 250, "extra metadata in Imp should not be removed")
+	assert.Equal(t, int(resultJSON.Imp[0].Video.W), 350, "extra metadata in Imp should not be removed")
+}
+
+func TestInvalidImpExt(t *testing.T) {
+	var a JWPlayerAdapter
+	a.endpoint = "http://test.com/openrtb2"
+
+	var reqInfo adapters.ExtraRequestInfo
+	reqInfo.PbsEntryPoint = "video"
+
+	request := &openrtb2.BidRequest{
+		ID: "test_id_1",
+		Imp: []openrtb2.Imp{{
+			ID: "test_imp_id",
+			Ext: json.RawMessage(`{]`),
+		}},
+	}
+
+	result, err := a.MakeRequests(request, &reqInfo)
+
+	assert.Len(t, err, 2, "2 errors should be returned")
+	assert.Empty(t, result, "Result should be nil")
+}
+
+func TestIdsAreRemoved(t *testing.T) {
+	var a JWPlayerAdapter
+	a.endpoint = "http://test.com/openrtb2"
+
+	var reqInfo adapters.ExtraRequestInfo
+	reqInfo.PbsEntryPoint = "video"
+
+	request := &openrtb2.BidRequest{
+		ID: "test_id",
+		Imp: []openrtb2.Imp{{
+			ID: "test_imp_id",
+			Ext: json.RawMessage(`{"bidder":{"placementId": "test_placement_id"}}`),
+		}},
+		Site: &openrtb2.Site{
+			ID: "test_site_id",
+			Domain: "test_domain",
+			Publisher: &openrtb2.Publisher{
+				ID: "test_publisher_id",
+				Name: "testPublisher_name",
+			},
+		},
+		App: &openrtb2.App{
+			ID: "test_app_id",
+			Domain: "test_app_domain",
+		},
+	}
+
+	results, err := a.MakeRequests(request, &reqInfo)
+
+	assert.Empty(t, err, "Errors array should be empty")
+	assert.Len(t, results, 1, "Only one request should be returned")
+
+	result := results[0]
+	resultJSON := &openrtb2.BidRequest{}
+	json.Unmarshal(result.Body, resultJSON)
+
+	assert.Len(t, resultJSON.Imp, 1, "Imp count should be equal or less than Imps from input. In this test, should be 1.")
+	assert.Empty(t, resultJSON.Imp[0].Ext, "Ext should be deleted")
+	assert.NotEmpty(t, resultJSON.Site, "Site object should not be removed")
+	assert.Empty(t, resultJSON.Site.ID, "Site.id should be removed")
+	assert.NotEmpty(t, resultJSON.Site.Publisher, "Publisher object should not be removed")
+	assert.Empty(t, resultJSON.Site.Publisher.ID, "Publisher.id should be removed")
+	assert.NotEmpty(t, resultJSON.App, "App object should not be removed")
+	assert.Empty(t, resultJSON.App.ID, "App.id should be removed")
+}
+
+
+func TestOpenRTBEmptyResponse(t *testing.T) {
+	httpResp := &adapters.ResponseData{
+		StatusCode: http.StatusNoContent,
+	}
+	bidder := new(JWPlayerAdapter)
+	bidResponse, errs := bidder.MakeBids(nil, nil, httpResp)
+
+	assert.Nil(t, bidResponse, "Expected empty response")
+	assert.Empty(t, errs, "Expected 0 errors. Got %d", len(errs))
+}
+
+func TestOpenRTBSurpriseResponse(t *testing.T) {
```

Comment:
do we have code coverage for BadServerResponse?

Reply:
Do you mean a 404 response ? Or a 200 with an invalid bid response ?

Reply:
not sure we have test coverage on this: